test/Microsoft.AspNetCore.SystemWebAdapters.Tests/HttpServerUtilityTests.cs

```diff
@@ -77,4 +86,53 @@ public void UrlTokenRoundtrip(string input, string expected)
         Assert.Equal(expected, encoded);
         Assert.Equal(bytes, decoded);
     }
+
+    // Test data from https://docs.microsoft.com/en-us/dotnet/api/system.web.httpserverutility.mappath?view=netframework-4.8
+    [InlineData("/RootLevelPage.aspx",null,"C:\\ExampleSites\\TestMapPath")]
+    [InlineData("/RootLevelPage.aspx", "", "C:\\ExampleSites\\TestMapPath")]
+    [InlineData("/RootLevelPage.aspx", "/DownOneLevel/DownLevelPage.aspx", "C:\\ExampleSites\\TestMapPath\\DownOneLevel\\DownLevelPage.aspx")]
+    [InlineData("/RootLevelPage.aspx", "/NotRealFolder", "C:\\ExampleSites\\TestMapPath\\NotRealFolder")]
+    [InlineData("/DownOneLevel/DownLevelPage.aspx", null, "C:\\ExampleSites\\TestMapPath\\DownOneLevel")]
+    [InlineData("/DownOneLevel/DownLevelPage.aspx", "../RootLevelPage.aspx", "C:\\ExampleSites\\TestMapPath\\RootLevelPage.aspx")]
+    [InlineData("/api/test/request/info", null, "C:\\ExampleSites\\TestMapPath\\api\\test\\request")]
+    [InlineData("/api/test/request/info", "", "C:\\ExampleSites\\TestMapPath\\api\\test\\request")]
+    [InlineData("/api/test/request/info", "/UploadedFiles", "C:\\ExampleSites\\TestMapPath\\UploadedFiles")]
+    [InlineData("/api/test/request/info", "UploadedFiles", "C:\\ExampleSites\\TestMapPath\\api\\test\\request\\UploadedFiles")]
+    [InlineData("/api/test/request/info", "~/MyUploadedFiles", "C:\\ExampleSites\\TestMapPath\\MyUploadedFiles")]
+    [Theory]
+    public void MapPath(string page, string? path, string expected)
```

Comment:
since the outcome here is expected to be a valid path, it makes sense that the tests are failing on linux/mac. 

What about changing the test to the following:

```suggestion
    public void MapPath(string page, string? path, params string[] expectedSegment)
```

Then using `Path.Combine` to construct the expected string?

Reply:
I've implemented this (as well as adjusting the `AppDomainAppPath` based on `System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(OSPlatform.Windows)`).
It would be really nice to be able to run the xunit tests using WSL and/or docker to be able to check the Linux tests on a windows machine. I think this is possible, but might require some setup to do.

Reply:
Woot - All checks passed!